src/game_config.cpp

```diff
@@ -91,17 +91,20 @@ bool
 const bool& debug = debug_impl;
 
 void set_debug(bool new_debug) {
+    // TODO: remove severity static casts and fix issue #7894
 	if(debug_impl && !new_debug) {
 		// Turning debug mode off; decrease deprecation severity
-		int severity;
+		lg::severity severity;
 		if(lg::get_log_domain_severity("deprecation", severity)) {
-			lg::set_log_domain_severity("deprecation", severity - 2);
+            int severityInt = static_cast<int>(severity);
+			lg::set_log_domain_severity("deprecation", static_cast<lg::severity>(severityInt - 2));
 		}
 	} else if(!debug_impl && new_debug) {
 		// Turning debug mode on; increase deprecation severity
-		int severity;
+        lg::severity severity;
 		if(lg::get_log_domain_severity("deprecation", severity)) {
-			lg::set_log_domain_severity("deprecation", severity + 2);
+            int severityInt = static_cast<int>(severity);
+			lg::set_log_domain_severity("deprecation", static_cast<lg::severity>(severityInt + 2));
```

Comment:
On the one hand, there's no range check here, and I think that might be Undefined Behavior https://www.open-std.org/jtc1/sc22/wg21/docs/cwg_defects.html#1766

OTOH, it's only reachable via command-line arguments, and I assume there will be a follow-up to this PR which addresses the TODO above. I'd be okay with merging as-is.

Reply:
IMO rather than a range check, just hardcoding them to specific log levels rather than doing math would be preferable. Doesn't need to be with this PR though.